### ISSUE TITLE
Jdbc Driver: java.sql.Statement.setMaxRows not implemented

### ISSUE THREAD

**Issue by lucboudreau** [prestodb/presto#3074]
This method doesn't do anything in terms of limiting the number of rows returned. 

The implementation should be the same as using the LIMIT clause in SQL.

Implementing this method is important for GUI tools. Many client tools will use a separate text box where users define the maximum number of rows to 'preview'. Since there is no ANSI SQL way to limit rows from SQL (yes, there is LIMIT, but that's rarely supported and non generic) using the Presto driver on a big database is risky and can make the client tools run out of memory very easily.

**Comment by electrum** [comment]
The simple fix is to implement this entirely in the `ResultSet`, though it would be preferable to pass this to the server so it can stop processing as soon as possible (even if the driver hasn't read all the results). We need to figure out a good way to do that (preferably without something hacky like rewriting the SQL in the driver).

**Comment by electrum** [comment]
We should add a new header `X-Presto-Max-Rows` rather than modifying the SQL.

**Comment by damiencarol** [comment]
@electrum I really need this feature. Do you know how to take down this one?

**Comment by fmeiser** [comment]
As far as I unterstand, we simply enable `StatementResource.createQuery` to read new header and pass the limit to method `QueryManager.createQuery`.
In method `QueryManager.createQuery`, before executing the statement, we set the limit in a copy of the statement

```
if (statement instanceof Query && limit > 0) {
	Query query = (Query) statement;
	statement = new Query(query.getLocation(), query.getQueryBody(), query.getOrderBy(), Integer.toString(limit), query.getApproximate());
}
```
@electrum Do you agree with the solution?

**Comment by electrum** [comment]
@fmeiser I think it's more complicated than that. There are other types of queries like `SHOW TABLES` that should also work for. The query modification probably needs to happen after analysis, otherwise we might return the wrong type of error for certain invalid queries.

@martint Where would you put this?

Start with the simple fix of implementing this in the driver's `ResultSet`. We'll want this anyway, to allow the feature to work with older server versions. Both `setMaxRows` and `setLargeMaxRows` should be implemented.

**Comment by fmeiser** [comment]
@electrum As a simple fix I implemented `setMaxRows` and `setLargeMaxRows` in `PrestoResultSet`

**Comment by fmeiser** [comment]
PR #5413 merged to master and released with 0.149
